shell/shared/admin.js

```diff
@@ -226,6 +226,14 @@ if (Meteor.isServer) {
         throw new Meteor.Error(403, "Unauthorized", "User must be admin or provide a valid token");
       }
 
+      // TODO(someday): currently this relies on the fact that an account is tied to a single
+      // identity, and thus has only that entry in "services". This will need to be looked at when
+      // multiple login methods/identities are allowed for a single account.
+      if (!value && !tokenIsValid(token) && (serviceName in Meteor.user().services)) {
+        throw new Meteor.Error(403, "Unauthorized",
+          "You can not disable the login service that your account uses.");
```

Comment:
The third argument to Meteor.Error() is hidden from the user in production mode, so users will only see "Unauthorized" here, which will be confusing. We actually make this mistake in a number of places. At one point I thought that the second argument was supposed to be an HTTP error string, but it's really supposed to be an arbitrary human-readable error description. Anyway, in this case just replace the second argument with the third and omit a third argument.